kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/NetworkTabsUi.java

```diff
@@ -245,73 +247,88 @@ public void setNetInterface(GwtNetInterfaceConfig selection) {
-        this.hardwareTab.setNetInterface(selection);
+        this.ip6Tab.setNetInterface(selection);
         this.dhcp4NatTab.setNetInterface(selection);
         this.wirelessTab.setNetInterface(selection);
         this.modemTab.setNetInterface(selection);
         this.modemGpsTab.setNetInterface(selection);
         this.modemAntennaTab.setNetInterface(selection);
+        this.hardwareTab.setNetInterface(selection);
 
-        // seems to be not needed
-        // this.visibleTabs.clear();
-        removeOptionalTabs();
         updateTabs();
         refreshAllVisibleTabs();
     }
 
-    private void removeOptionalTabs() {
-        this.tabsPanel.remove(this.wirelessTabAnchorItem);
-        this.tabsPanel.remove(this.modemTabAnchorItem);
-        this.tabsPanel.remove(this.dhcp4NatTabAnchorItem);
+    public void updateTabs() {
+        removeAllTabs();
+
+        insertTab(this.ip4TabAnchorItem);
+        if (this.isNet2) {
+            insertTab(this.ip6TabAnchorItem);
+        }
+
+        arrangeOptionalTabs();
+
+        insertTab(this.hardwareTabAnchorItem);
+    }
+
+    private void removeAllTabs() {
+        removeTab(this.ip4TabAnchorItem);
+        removeTab(this.ip6TabAnchorItem);
+        removeTab(this.wirelessTabAnchorItem);
+        removeTab(this.dhcp4NatTabAnchorItem);
+        removeTab(this.modemTabAnchorItem);
+        removeTab(this.modemGpsTabAnchorItem);
+        removeTab(this.modemAntennaTabAnchorItem);
+        removeTab(this.hardwareTabAnchorItem);
     }
 
-    public void updateTabs() {
-        getCurrentConfiguration();
+    private void arrangeOptionalTabs() {
+        boolean isIpv4EnabledLAN = this.ip4Tab.getStatus().equals(IPV4_STATUS_ENABLED_LAN_MESSAGE);
+        boolean isIpv4Disabled = this.ip4Tab.getStatus().equals(IPV4_STATUS_DISABLED_MESSAGE);
+        boolean isIpv4Unmanaged = this.ip4Tab.getStatus().equals(IPV4_STATUS_UNMANAGED_MESSAGE);
+        boolean isWirelessAP = this.wirelessTab.getWirelessMode() != null
+                && this.wirelessTab.getWirelessMode().name().equals(WIFI_ACCESS_POINT);
+        boolean isDhcp = this.ip4Tab.isDhcp();
 
-        boolean includeDhcpNat = !this.isDhcp && this.isIpv4EnabledLAN;
+        boolean includeDhcpNat = !isDhcp && isIpv4EnabledLAN;
 
-        if (this.netIfConfig instanceof GwtWifiNetInterfaceConfig) {
+        InterfaceConfigWrapper wrapper = new InterfaceConfigWrapper(this.netIfConfig);
+
+        if (wrapper.isWireless()) {
             showWirelessTabs();
-            if (!this.isWirelessAP) {
+            if (!isWirelessAP) {
                 includeDhcpNat = false;
             }
-        } else if (this.netIfConfig instanceof GwtModemInterfaceConfig) {
+        } else if (wrapper.isModem()) {
             includeDhcpNat = false;
-            this.modemGpsTabAnchorItem.setEnabled(this.isGpsSupported && !this.isIpv4Unmanaged);
+            this.modemGpsTabAnchorItem.setEnabled(wrapper.isGpsSupported() && !isIpv4Unmanaged);
```

Comment:
The behavior here (as the same of the previous one) is not to show the gps tab if the modem is unmanaged. Is it the expected behavior? Probably this is out of the scope of this PR, but it worth to discuss on it.

Reply:
I agree we should discuss it further but outside this PR.